db/helpers/generatorUtils.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * db/helpers/generatorUtils.js
+ *
+ * DB utils for the generator model.
+ */
+const crypto = require('crypto');
+const dbConstants = require('../constants');
+
+/**
+ * Encrypt the given data using passed in secretKey and algorithm
+ * @param  {String} data -  Data to be encrypted
+ * @param  {String} secretKey - Secret key for encryption
+ * @param  {String} algorithm - Name of the encryption algorithm
+ * @returns {String} - encrypted data
+ */
+function encrypt(data, secretKey, algorithm) {
+  const cipher = crypto.createCipher(algorithm, secretKey);
+  let crypted = cipher.update(data, 'utf-8', 'hex');
+  crypted += cipher.final('hex');
+  return crypted;
+} // encrypt
+
+/**
+ * Decrypt encrypted data using passed in secretKey and algorithm
+ * @param  {String} data -  Data to be decrypted
+ * @param  {String} secretKey - Secret key that was used from encryption
+ * @param  {String} algorithm - Name of the encryption algorithm
+ * @returns {String} - encrypted data
+ */
+function decrypt(data, secretKey, algorithm) {
+  const decipher = crypto.createDecipher(algorithm, secretKey);
+  let decrypted = decipher.update(data, 'hex', 'utf-8');
+  decrypted += decipher.final('utf-8');
+  return decrypted;
+} // decrypt
+
+/**
+ * This function fetches the sample generator secret key and the sample
+ * generator encryption algorithm from the glocalConfig model
+ * and returns an object containing the secretKey and algorithm
+ * @param {Model} globalConfigModel - The sequelize Global Config Model
+ * @returns {Object} - containing the secret key and algorithm
```

Comment:
what happens if either key does not exist? throw an error and reject creating a sample generator if its template specifies encryption for any of its context vars?

Reply:
I think we talked about this, and we said that we will not do the encryption if the keys do not exist.

Reply:
Oh, I forgot. Can you add comments in the code here to make it very clear?

Also, let's get this information added to https://salesforce.github.io/refocus/docs/10-security.html (separate PR OK)

Reply:
Sure, will do.

Reply:
Is there a reason why we chose 'no encryption' over 'throwing error' ?
Basically, we are saying that the context vars will be in plain text even if they have been specified to be "encrypted" ?

Reply:
If there no secret key and algorithm in the global config, the context variables will be stored in plain text.

Reply:
umm okay. I guess as a user I will not be happy if my secured variables are stored as plain text because of admin's decision. But I am assuming you have discussed it, so its the best solution as of now.

Reply:
thx for your comment @pallavi2209 I think we should reconsider that
@shriramshankar can you ask Iain and Charlie what they think?

Reply:
This is more or less like the "require https" feature we have. Yes, security is really important and we should have everything in place for a secure installation. But it should be made configurable and left to the Admin to decide how and what level of security is needed for their installation.

Reply:
@meynet @iusiri  What are your thoughts about this? 
When a context variable is required to be encrypted but the Admin has not yet provided the secretKey and algorithm in the global config table. Should the post request, to add the generator to refocus be rejected?

Reply:
Iain thinks throwing an error is better. Let's see what Charlie says and we can do what most of us think is right to do.

Reply:
I am fine with either case after you have mentioned "require https" explanation. If we do not get his reply soon, we can probably merge this (I am waiting on this one) and create a new PR if we need to do any further change.